docs/common/pki-cli.md

```diff
@@ -0,0 +1,456 @@
+```
+
+Note that the DN will be overriden by the OSG PKI except for the CN component.
+
+Submitting the request:
+
+```console
+[root@client ~] # osg-cert-request -t hostname.domain.com -e emailaddress@domain.com -n "Your Name" -p 9999999999 -y "xyz@domain.com,abc@domain.com" -m "This is my comment" --csr csr.pem
+```
+
+### osg-cert-retrieve
+
+Retrieve a certificate (host or user) from OIM given a request Id. Typically you will run this script after submitting a request with `osg-cert-request` and receiving an email telling you your certificate has been approved.
+
+You can also use this script to retrieve other certificates that have been previously issued (assuming you know their request ID number).
+
+Since certificates are public, no authentication of the user is required.
+
+This script:
+
+-   Accepts a request Id from the user
+-   Connects to OIM and requests the certificate identified by the request ID
+-   Writes the certificate to disk (as specified by the user)
+
+**Inputs:**
+
+-   Request ID
+-   Filename to store certificate (optional: default is `./hostcert.pem`).
+
+**Outputs:**
+
+-   Host certificate as PEM, to filename specified or `./hostcert.pem`.
+
+```console
+[root@client ~] # osg-cert-retrieve --help
+Usage: osg-cert-retrieve [options] <Request ID>
+Usage: osg-cert-retrieve -h/--help [for detailed explanations of options]
+
+Options:
+  -h, --help            show this help message and exit
+  -o ID, --certfile=ID  Specify the output filename for the retrieved user
+                        certificate . Default is ./hostcert.pem
+  -T, --test            Run in test mode
+  -t TIMEOUT, --timeout=TIMEOUT
+                        Specify the timeout in minutes
+  -q, --quiet           don't print status messages to stdout
+  -d WRITE_DIRECTORY, --directory=WRITE_DIRECTORY
+                        Write the output files to this directory
+  -V, --version         Print version information and exit
+  -i, --id              Specify ID# of certificate to be retrieved
+                        [deprecated]
+```
+
+Example:
+
+```console
+[root@client ~] # osg-cert-retrieve -i 555
+```
+
+### osg-gridadmin-cert-request
+
+Request and retrieve multliple host certificates from OIM. Authenticates to OIM and is only for use by Grid Admins for certificates they are authorized to approve. This script is only supported with all hosts being in the same domain (so we ensure they go to the same Grid Admin). The certificates are stored with the format of `hostname-requestid.pem` (i.e. the id generated from the request for the certificate). The key is stored as `hostname-serial-key.pem`.
+
+This script does the following in the process of acquiring certificates for the hostnames specified:
+
+-   Reads a list of fully-qualified hostnames from a file specified by the user.
+-   For each hostname:
+    -   Generates a new private key and `CSR`.
```

Comment:
That's not what I meant. "CSR" shouldn't be monospace here at all since it's not monospace anywhere else.